### ISSUE TITLE
will this project support windows?

### ISSUE THREAD

**Issue by hh9527** [ticki/termion#61]
Like what https://github.com/nsf/termbox-go has done.

**Comment by ticki** [comment]
Termbox is different API-wise in that it's high level. Termion tries to be as low-level as possible (for example, you can write the escape codes directly into a string and output it later).

If you put the Windows API calls into the `fmt::Display`s of the various structs, you get problems with caching: the call will output instantly, which might not align to the output stream.

So it is simply not possible as it stands. You need a higher-level API (which comes on the price of performance and expressiveness).

There are good news, however. I believe Microsoft are planning to support ANSI codes, which we can only hope for. The current approach is not very well-designed.

In conclusion, there is no way we can support it currently.

**Comment by ziotom78** [comment]
> There are good news, however. I believe Microsoft are planning to support ANSI codes, which we can only hope for. The current approach is not very well-designed.

For future readers interested in this matter, here is a relevant link: https://blogs.msdn.microsoft.com/commandline/2016/09/22/24-bit-color-in-the-windows-console/

In particular, see the answer the author (Rich Turner) gave to user "belleve":

> Belleve: How to use them in Win32 API? I mean, using a new API or use escape sequences?
> Rich Turner: Our advice is to use VT sequences for access to the new, advanced features of the console.

If Microsoft is going to stick close to xterm's escape sequences (and there is no reason to believe they will not), it might be that Termios will support Windows for free in a near future. (People running the latest Windows Insider Build might check if this is already the case.)

**Comment by ticki** [comment]
@ziotom78 Sounds very cool.

**Comment by ticki** [comment]
ping @retep998

**Comment by retep998** [comment]
If you want to support the new VT sequence support in Windows, all you have to do is set the appropriate console mode via `SetConsoleMode`, detect what features are available based on the version of Windows 10 you're on, and then have fun.
